Allow kickscooters on roads with low speed limit

AdamWiktor opened the pull request.

AdamWiktor committed:
Allow kickscooters on roads with low speed limit

vessla commented on src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java:
```diff
@@ -223,16 +244,20 @@ public boolean canTraverse(VehicleDescription vehicle) {
+                    if (getMaxStreetTraverseSpeed() > KickScooterDescription.MAX_EDGE_TRAVERSE_SPEED_LOWER_BOUND) {
+                        //If maximum traverse speed > 30 km/h, kickscooters are not allowed
+                        return false;
+                    }
```
It seems that these changes alter the paths returned in response to the requests for some integration tests: 

ResponseItineraryTypeBasicIT.testBusAndKickscooter:98 
Expected: iterable containing ["WALK+KICKSCOOTER", "WALK+TRANSIT", "WALK+TRANSIT"]
but: item 0: was "WALK+TRANSIT"

ResponseItineraryTypeBasicIT.testKickscooter:73 » Processing Error reading ent...

vessla commented on src/main/java/org/opentripplanner/routing/core/vehicle_sharing/KickScooterDescription.java:
```diff
@@ -7,11 +7,20 @@
+     so we assume kickscooters are a bit faster on bikepaths than on streets.
+     */
+    protected static final double MAX_SPEED_IN_METERS_PER_SECOND_ON_BIKEPATH = 20. * (10. / 36.);
+    protected static final double MAX_SPEED_IN_METERS_PER_SECOND_ON_ROAD = 19. * (10. / 36.);
```
It seems that these changes alter the paths returned in response to the requests for some integration tests: 

ResponseItineraryTypeBasicIT.testBusAndKickscooter:98 
Expected: iterable containing ["WALK+KICKSCOOTER", "WALK+TRANSIT", "WALK+TRANSIT"]
but: item 0: was "WALK+TRANSIT"

ResponseItineraryTypeBasicIT.testKickscooter:73 » Processing Error reading ent...

AdamWiktor replied:
It turns out that pedestrian and bike paths doesn't have proper max speed set (this field doesn't make sense on those paths obviously). Fixed

AdamWiktor committed:
Fix speed lower bound check for kickscooters

Pedestrian and bike paths doesn't have max traverse speed set correctly,
so we should ignore max speed lower bound in that case.

AdamWiktor committed:
Merge branch 'dev-1.x' into feature/VMP-192-allow-kickscooters-on-roads-with-lowspeed-limit
